app/controllers/support_interface/sessions_controller.rb

```diff
@@ -4,12 +4,46 @@ class SessionsController < SupportInterfaceController
 
     def new
       session['post_dfe_sign_in_path'] ||= support_interface_path
+      if FeatureFlag.active?('dfe_sign_in_fallback')
+        render :authentication_fallback
+      end
     end
 
     def destroy
       DfESignInUser.end_session!(session)
 
       redirect_to action: :new
     end
+
+    def sign_in_by_email
+      render_404 and return unless FeatureFlag.active?('dfe_sign_in_fallback')
+
+      support_user = SupportUser.find_by(email_address: params.dig(:support_user, :email_address).downcase.strip)
+
+      if support_user
+        SupportInterface::MagicLinkAuthentication.send_token!(support_user: support_user)
+      end
+
+      redirect_to support_interface_check_your_email_path
+    end
+
+    def authenticate_with_token
+      redirect_to action: :new and return unless FeatureFlag.active?('dfe_sign_in_fallback')
+
+      support_user = SupportInterface::MagicLinkAuthentication.get_user_from_token!(token: params.fetch(:token))
```

Comment:
This could use https://github.com/DFE-Digital/apply-for-teacher-training/pull/2362 as well.